server/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -200,7 +200,8 @@ public static void parseSearchRequest(
         searchRequest.preference(request.param("preference"));
         searchRequest.indicesOptions(IndicesOptions.fromRequest(request, searchRequest.indicesOptions()));
 
-        checkRestTotalHits(request, searchRequest);
+        validateSearchRequest(request, searchRequest);
```

Comment:
If `validateSearchRequest` already includes `checkSearchType` why do we need to call it again after `validateSearchRequest`?

Reply:
Thanks @mayya-sharipova , this was just an oversight. Will fix.